src/Propel/Common/Util/CartesianProduct.php

```diff
@@ -0,0 +1,43 @@
+<?php namespace Propel\Common\Util;
+
+class CartesianProduct
```

Comment:
I would like to see some way to limit the size of this. Cartesian products end up being expensive for both CPU and RAM in easily process-crashing-ways.